contracts/Raffles.cdc

```diff
@@ -85,18 +98,36 @@ pub contract Raffles {
             return self.addresses[index]
         }
 
+        pub fun getAddresses(): [Address] {
+            return self.addresses
+        }
+
+        pub fun addAddress(address: Address) {
+            self.addresses.append(address)
+        }
+
         init() {
             self.addresses = []
         }
     }
 
     pub resource interface ManagerPublic {
         pub fun borrowRafflePublic(id: UInt64): &{RafflePublic}?
+        pub fun getIDs(): [UInt64]
+        pub fun getRaffleAddresses(id: UInt64): [Address]
     }
 
     pub resource Manager: ManagerPublic {
         access(self) let raffles: @{UInt64: Raffle}
 
+        pub fun getIDs(): [UInt64] {
+            return self.raffles.keys
+        }
+
+        pub fun getRaffleAddresses(id: UInt64): [Address] {
+            return (&self.raffles[id] as &Raffle?)!.getAddresses() 
+        }
```

Comment:
Generally with method like this, they should exist in the underlying resource you are calling, and a borrow method should be used to get to it (`borrowRafflePublic` in this case)